### PR TITLE
Add GranteeService.getVisibleUsers()

### PR DESCRIPTION
**Pull request by mpfz0r**
Graylog users are visible to other users if the user that is running
the query has `users:read` permissions on certain (or all) users.

The `users:read` permission can be conveniently assigned by using the
new `User Inspector` role.

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise#1833

### CODE REVIEW

File: graylog2-server/src/test/java/org/graylog/testing/UserServiceExtension.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return Objects.equals(UserService.class, parameterContext.getParameter().getType());
+    }
+
+    @Override
+    public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        final Class<?> parameterType = parameterContext.getParameter().getType();
+
+        if (UserService.class.equals(parameterType)) {
```
Comment (bernd) [inline graylog2-server/src/test/java/org/graylog/testing/UserServiceExtension.java:41]: Should we only inject `TestUserService` instances? That would make it clear that the user of the extension gets a `TestUserService` and not a fully featured `UserServiceImpl`. (we might want to rename the extension to `TestUserServiceExtension` in that case, too)

```suggestion
        if (TeamUserService.class.equals(parameterType)) {
```

**Review by bernd — CHANGES_REQUESTED** [review]
LGTM :+1: Only one question about the `UserServiceExtension`

File: graylog2-server/src/test/java/org/graylog/testing/UserServiceExtension.java (hunk 1 above)
Comment (mpfz0r) [reply graylog2-server/src/test/java/org/graylog/testing/UserServiceExtension.java]: Makes sense :+1:

File: graylog2-web-interface/src/logic/permissions/GRN.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ export const getShowRouteFromGRN = (grn: string) => {
       return Routes.dashboard_show(id);
     case 'event_definition':
       return Routes.ALERTS.DEFINITIONS.edit(id);
-    case 'event_notification':
+    case 'notification':
```
Comment (bernd) [inline graylog2-web-interface/src/logic/permissions/GRN.js:28]: Is this change intended?

**Review by bernd — CHANGES_REQUESTED** [review]

File: graylog2-web-interface/src/logic/permissions/GRN.js (hunk 2 above)
Comment (mpfz0r) [reply graylog2-web-interface/src/logic/permissions/GRN.js:28]: oops, no. but after a merge it will be gone. It's in master already

Comment (bernd) [reply graylog2-web-interface/src/logic/permissions/GRN.js:28]: Okay :+1:

**Review by bernd — APPROVED** [review]
